chapter_10/10-8.py

```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+
+pet_names = ['./chapter_10/cats.txt', './chapter_10/dogs.txt']

```

Comment:
Your code would be more robust if you used a relative path to the tune of
pet_names = ['cats.txt', 'dogs.txt']

Or if you anchored the path, to the tune of what Alexa did in her chapter 10-12 (this wouldn't be the exact code but it's an example of how she roots her directory)

import json
import os
from pathlib import Path

def getRootDir():
    """ """
    # deal with "What folder am I working from?"
    cwd = os.getcwd()
    index = cwd.find("chapter")
    if index > 0:
        cwd = cwd[:index]
    # else do nothing, assume we're in workspace root
    return cwd

def main():
    """ """
    rootDir = getRootDir()

    # create scratch dir
    scratchDir = Path(f"{rootDir}/scratch_files")
    if not scratchDir.exists():
        os.mkdir(scratchDir.resolve())

This change still leaves lots of potential failure cases when a user pulls the code(I'm still getting an error where I shouldn't, and I am in the PCC directory but its not in the same base location as you). At the very least you should comment the required filepath/placement somewhere in here so a developer would know how to set it up.